compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -577,15 +576,24 @@ static LogicalResult setTransposeConfig(func::FuncOp entryPoint,
   int32_t tileM = 32;
   int32_t tileN = 32;
   TileSizesListType tileSizes;
-  tileSizes.push_back({tileM, tileN});
 
-  // Check alignment with tile size for each transpose.
+  auto linalgOp = dyn_cast<linalg::GenericOp>(op);
+  if (!linalgOp) {
+    return failure();
+  }
+  // Set all tile sizes to 1 except for M & N
+  SmallVector<int64_t> tileSizesTemp(linalgOp.getNumLoops() - 2, 1);
+  tileSizesTemp.push_back(tileM);
+  tileSizesTemp.push_back(tileN);
```

Comment:
this assumes the permutation is done between the last two dimensions (i.e. (d0, d1, d2) -> (d0, d2, d1) map). What if the map is  (d0, d1, d2) -> (d2, d1, d0) instead? 
We should make sure this code handles efficiently all the cases, and we should also consider cases where different operands have different permutations.

Reply:
Good point. I don't actually see a way to handle different ops having different permutations. By the time we reach this point the operations have already been fused together. I feel like we'd need separate loops to handle that?

Reply:
Handling different operands with different permutations is hard and probably we would want to skip. We should be able to handle cases where all the inputs have the same most inner dimension (basically each input needs to have inner dimension be either d2 or the same other dimension. But we still need to pick the right dimension to tile in this case.